src/ZendeskApi.Contracts/Models/User.cs

```diff
@@ -92,5 +92,8 @@ public class User : IZendeskEntity
 
         [DataMember(Name = "ticket_restriction", EmitDefaultValue = false)]
         public string TicketRestriction { get; set; }
+
+        [DataMember(Name = "user_fields", EmitDefaultValue = false)]
+        public Dictionary<string, object> UserFields { get; set; }
```

Comment:
From https://developer.zendesk.com/rest_api/docs/core/users#json-format-for-agent-or-admin-requests, the value would be a `hash` or in .NET terms, a dictionary with string keys and arbitrary types for values.

Here's a snippet from Zendesk API.

```
"user_fields": {
    "user_decimal": 5.1,
    "user_dropdown": "option_1",
    "user_date": "2012-07-23T00:00:00Z"
  }
```

I also haven't done much with user fields except a simple string key/value which lead me to submit this PR. I won't have enough time to explore user fields any further to come up with a general solution that will fit with the other implementations.

I recommend this be added as a dictionary until we have a better understanding of what the implementation would look like.